demos/2024-04-23-cheritech/demo.cc

```diff
@@ -0,0 +1,366 @@
+// Copyright SCI Semiconductor and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+#include <NetAPI.h>
+#include <cstdlib>
+#include <debug.hh>
+#include <errno.h>
+#include <fail-simulator-on-error.h>
+#include <locks.hh>
+#include <mqtt.h>
+#include <platform-entropy.hh>
+#include <platform-gpio.hh>
+#include <sntp.h>
+#include <tick_macros.h>
+
+#include "host.cert.h"
+#include "thread.h"
+#include "timeout.h"
+//#include "mosquitto.org.h"
+
+using CHERI::Capability;
+
+using Debug            = ConditionalDebug<true, "MQTT demo">;
+constexpr bool UseIPv6 = CHERIOT_RTOS_OPTION_IPv6;
+
+// MQTT network buffer sizes
+constexpr const size_t networkBufferSize    = 1024;
+constexpr const size_t incomingPublishCount = 100;
+constexpr const size_t outgoingPublishCount = 100;
+
+namespace
+{
+
+	DECLARE_AND_DEFINE_CONNECTION_CAPABILITY(DemoHost,
+	                                         "cheriot.demo",
+	                                         8883,
+	                                         ConnectionTypeTCP);
+
+	DECLARE_AND_DEFINE_ALLOCATOR_CAPABILITY(mqttTestMalloc, 32 * 1024);
+
+	constexpr const char *ledTopic             = "cheri-led";
+	int32_t               ledSubscribePacketId = -1;
+	bool                  ledAckReceived       = false;
+
+	constexpr const char *buttonTopic   = "cheri-button";
+	int                   buttonCounter = 0;
+
+	/// Helpers
+
+	/// Returns a weak pseudo-random number.
+	uint64_t rand()
+	{
+		EntropySource rng;
+		return rng();
+	}
+
+	/**
+	 * Note from the MQTT 3.1.1 spec:
+	 * The Server MUST allow ClientIds which are between 1 and 23 UTF-8 encoded
+	 * bytes in length, and that contain only the characters
+	 * "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	 *
+	 * Note from us:
+	 * UTF-8 encoding of 0-9, a-z, A-Z, is 1 Byte per character, so we should be
+	 * able to do up to a length of 22 characters + zero byte.
+	 */
+	constexpr const int clientIDlength = 23;
+	constexpr const int clientIDPrefixLength = 8;
+	char                clientID[clientIDlength] = "cheriot-XXXXXXXXXXXXXX";
+
+	/**
+	 * Turn an LED on.
```

Comment:
Should this comment be different from the one for `led_on`?

Reply:
You seem to be reviewing something that was removed in a later commit.